tests/api/users_test.py

```diff
@@ -31,7 +32,107 @@ def test_create_attributes(self):
             auth=client.auth,
             headers={
-                'Content-Type': 'application/json',
-                'User-Agent': f'Stytch Python v{version}'
+                "Content-Type": "application/json",
+                "User-Agent": f"Stytch Python v{version}",
             },
         )
+
+
+class TestSearchUsers(unittest.TestCase):
+    def test_search_no_params(self):
+        client = FakeClient()
+        response = FakeResponse(
+            status_code=200,
+            body='{"user_id": "user-test-183e939c-e7e1-4d56-82da-8ba9c6036878"}',
+        )
+
+        with mock.patch.object(requests, "post", return_value=response) as mock_post:
+            users = Users(client)
+            users._requester_base = requests
+            _ = users.search()
+
+        mock_post.assert_called_once_with(
+            "https://localhost:8080/users/search",
+            data="{}",
+            auth=client.auth,
+            headers={
+                "Content-Type": "application/json",
+                "User-Agent": f"Stytch Python v{version}",
+            },
+        )
+
+    def test_search_with_params(self):
+        client = FakeClient()
+        response = FakeResponse(
+            status_code=200,
+            body='{"user_id": "user-test-183e939c-e7e1-4d56-82da-8ba9c6036878"}',
+        )
+
+        with mock.patch.object(requests, "post", return_value=response) as mock_post:
+            users = Users(client)
+            users._requester_base = requests
+            _ = users.search(
+                limit=50,
+                cursor="abc123",
+                query={
+                    "operator": "AND",
+                    "operands": [{"filter_name": "test_filter", "filter_value": 1}],
+                },
+            )
+
+        mock_post.assert_called_once_with(
+            "https://localhost:8080/users/search",
+            data='{"limit": 50, "cursor": "abc123", "query": {"operator": "AND", "operands": [{"filter_name": "test_filter", "filter_value": 1}]}}',
+            auth=client.auth,
+            headers={
+                "Content-Type": "application/json",
+                "User-Agent": f"Stytch Python v{version}",
+            },
+        )
+
+    @mock.patch("requests.post")
+    def test_search_all(self, mock_post):
+        client = FakeClient()
+        response_1 = FakeResponse(
+            status_code=200,
+            body='{"results_metadata": {"next_cursor": "A"}}',
+        )
+
+        response_2 = FakeResponse(
+            status_code=200,
+            body='{"results_metadata": {"next_cursor": null}}',
+        )
+
+        users = Users(client)
+        users._requester_base = requests
+
+        mock_post.return_value = response_1
+        search_generator = users.search_all()
+
+        _ = next(search_generator)
```

Comment:
While we're here, it might be nice to show the format that the generator yields. If I understand it correctly:

```python
page = next(search_generator)
assert page == [{...}]
```